drivers/regulator/regulator_npm1300.c

```diff
@@ -670,9 +675,9 @@ static DEVICE_API(regulator, api) = {
 		.source = _source,                                                                 \
 		.retention_uv = DT_PROP_OR(node_id, retention_microvolt, 0),                       \
 		.soft_start = DT_ENUM_IDX_OR(node_id, soft_start_microamp, UINT8_MAX),             \
-		.enable_gpios = GPIO_DT_SPEC_GET_OR(node_id, enable_gpios, {0}),                   \
-		.retention_gpios = GPIO_DT_SPEC_GET_OR(node_id, retention_gpios, {0}),             \
-		.pwm_gpios = GPIO_DT_SPEC_GET_OR(node_id, pwm_gpios, {0}),                         \
+		.enable_gpios = DT_PROP_OR(node_id, enable_gpio_config, {UINT8_MAX}),              \
+		.retention_gpios = DT_PROP_OR(node_id, retention_gpio_config, {UINT8_MAX}),        \
+		.pwm_gpios = DT_PROP_OR(node_id, pwm_gpio_config, {UINT8_MAX}),                    \
```

Comment:
how about instead of hardcoding `{UINT8_MAX}` here, put a define with the default next to the struct definition and set both field? something like `#define NPM1300_GPIO_INFO {0xff, 0x00}`

Reply:
Switched `UINT8_MAX` to a define